plugins/metadata/MetadataPlugin.php

```diff
@@ -679,9 +684,14 @@ protected static function addMetadataWithProfilesSystemNames (BaseObject $object
     		KalturaLog::info("Metadata [" . $dbMetadata->getId() . "] saved [$xmlData]");
     		
     		$key = $dbMetadata->getSyncKey(Metadata::FILE_SYNC_METADATA_DATA);
-    		kFileSyncUtils::file_put_contents($key, $xmlData);
-    		
-		    kEventsManager::raiseEvent(new kObjectDataChangedEvent($dbMetadata));
+    		if (!kFileSyncUtils::compareContent($key, $xmlData))
```

Comment:
Bettern to move this section into a new function - private saveMetadataFile( $dbMetadata,$xmlData) and use it in the 2 locations